src/bun.js/bindings/JSReadableHelper.cpp

```diff
@@ -222,7 +175,76 @@ JSC_DEFINE_HOST_FUNCTION(jsReadable_emitReadable_, (JSGlobalObject * lexicalGlob
 
     state->m_needReadable = state->m_flowing <= 0 && !state->m_ended && state->m_length <= state->m_highWaterMark;
     flow(lexicalGlobalObject, stream, state);
+}
+
+JSC_DECLARE_HOST_FUNCTION(jsReadable_emitReadable_);
+JSC_DEFINE_HOST_FUNCTION(jsReadable_emitReadable_, (JSGlobalObject * lexicalGlobalObject, CallFrame* callFrame))
+{
+    JSReadableHelper_EXTRACT_STREAM_STATE
+
+    emitReadable_(lexicalGlobalObject, stream, state);
+
+    RELEASE_AND_RETURN(throwScope, JSValue::encode(jsUndefined()));
+}
+
+void emitReadable(JSGlobalObject* lexicalGlobalObject, JSObject* stream, JSReadableState* state)
+{
+    VM& vm = lexicalGlobalObject->vm();
+
+    state->m_needReadable = false;
+    if (!state->m_emittedReadable) {
+        state->m_emittedReadable = true;
+        // make this static?
+        JSFunction* emitReadable_ = JSC::JSFunction::create(
```

Comment:
Maybe have a StreamsModule struct which has this function as a pointer and have global object store a pointer to it (null by default)?

Reply:
fixed.